### ISSUE TITLE
Can't open links in Firefox

### ISSUE THREAD

**Issue by TheToto** [simoniz0r/Discord-AppImage#4]
Hello,
(I'm on ArchLinux)
When i try to open a link from Discord (with firefox as default browser), it doesn't work ;
Because after setting LD_LIBRARY_PATH to appimage lib folders :
```
$ xdg-open https://google.fr           
XPCOMGlueLoad error for file /usr/lib/firefox/libxul.so:                                                                                     
/home/toto/discord/squashfs-root/usr/lib/x86_64-linux-gnu/libssl3.so: version `NSS_3.33' not found (required by /usr/lib/firefox/libxul.so)
Couldn't load XPCOM.
```
So Discord can't open links because firefox fails to open with appimage libs.
Some tips/workaround ?

**Comment by simoniz0r** [comment]
I cannot reproduce this.  Make sure Firefox is set as your default browser in xdg-settings.  Setting the `LD_LIBRARY_PATH` has nothing to do with opening a browser.

**Comment by EthanWeegee** [comment]
Can reproduce. A popup opens, saying `Failed to execute default Web Browser. Input/output error.` Opening links works correctly with the `discord-installer` package. Running `xdg-open www.example.com` opens a tab in Firefox as expected.

**Comment by Exagone313** [comment]
I reproduce this issue too. Will see if there is a way to make some kind of wrapper than unset some environment variables.

**Comment by Exagone313** [comment]
I can confirm that unsetting `LD_LIBRARY_PATH` in an `xdg-open` wrapper will fix the issue.

There are probably more informations on this GitHub issue: https://github.com/NixOS/nixpkgs/issues/160923

I do not use nixOS, but it probably has more in depth informations. For example, it seems to happen with some other appimages too.

For the record, here is the value I get for this environment variable:

```
LD_LIBRARY_PATH=/tmp/.mount_discorSPl0X1/usr/lib/:/tmp/.mount_discorSPl0X1/usr/lib/x86_64-linux-gnu:
```

And an example wrapper:

```bash
#!/bin/sh
# discord sets this variable which breaks wayland apps
unset GDK_BACKEND
# discord app image sets this
if [ -n "${APPDIR}" ]; then
	unset LD_LIBRARY_PATH
fi
exec /usr/bin/xdg-open "$@"
```

**Comment by junetried** [comment]
> I can confirm that unsetting `LD_LIBRARY_PATH` in an `xdg-open` wrapper will fix the issue.
> 
> There are probably more informations on this GitHub issue: [NixOS/nixpkgs#160923](https://github.com/NixOS/nixpkgs/issues/160923)
> 
> I do not use nixOS, but it probably has more in depth informations. For example, it seems to happen with some other appimages too.
> 
> For the record, here is the value I get for this environment variable:
> 
> ```
> LD_LIBRARY_PATH=/tmp/.mount_discorSPl0X1/usr/lib/:/tmp/.mount_discorSPl0X1/usr/lib/x86_64-linux-gnu:
> ```
> 
> And an example wrapper:
> 
> ```shell
> #!/bin/sh
> # discord sets this variable which breaks wayland apps
> unset GDK_BACKEND
> # discord app image sets this
> if [ -n "${APPDIR}" ]; then
> 	unset LD_LIBRARY_PATH
> fi
> exec /usr/bin/xdg-open "$@"
> ```

This isn't an acceptable fix, but it does suggest that libraries are at fault for this issue, which I've suggested before but it was quickly shot down.

**Comment by simoniz0r** [comment]
After thinking about this, the only things I can think of that are bundled that could cause this are `libnspr4` and `libnss3` as those are both used by Firefox.  I will try building three test AppImages to see if one or both of those are the issue.

**Comment by simoniz0r** [comment]
This should be fixed by 0d7929b.  Please try redownloading the AppImage from the [releases](https://github.com/simoniz0r/Discord-AppImage/releases) page and let me know if this does not fix the issue for you.
